src/app/donation-start/donation-start-form/PaymentReadinessTracker.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * This object tracks whether a donor is ready to progress past the payment stage of the donation start form. At time of
+ * writing it just returns a boolean from readyToProgressFromPaymentStep, but my plan is to add a function to return
+ * a list of reasons *why* we are not ready for them to proceed that we can display using snackbar.
+ *
+ * Originally extracted from the DonationStartFormComponent
+ */
+export class PaymentReadinessTracker {
+  /**
+   * Property name taken from the component. The name doesn't really reflect the usage. Planning to remove this property.
+   */
+  private stripePaymentMethodReady: boolean = false;
```

Comment:
these properties are copies of the properties in DonationStartFormComponent. Every time one of the matching properties in DonationStartFormComponent was set previously there should now be a call to a function on this object that sets the property here.

Tomorrow I'll probably at least one of them, maybe more and replace with more specific properties that say exactly what the relevant state is, so we can generate not only the boolean answer but also the list of reasons for a no.

Reply:
@bdsl I'm not clear what's next here, I think there's a typo in
> I'll probably at least one of them

Reply:
should have said "probably remove" at least one of them. But I've already removed `stripePaymentMethodReady` after I wrote that comment.